gradle/validate-annotations.gradle

```diff
@@ -0,0 +1,44 @@
+for (def sourceSet in [sourceSets.main, sourceSets.client]) {
+	// We have to capture the source set name for the lazy string literals,
+	// otherwise it'll just be whatever the last source set is in the list.
+	def sourceSetName = sourceSet.name
+	def taskName = sourceSet.getTaskName('validate', 'Annotations')
+	def task = tasks.register(taskName, ValidateAnnotations) {
+		group = 'fabric'
+		description = "Validate annotations for $sourceSetName packages."
+
+		// Only apply to default source directory since there's also generated package-info files.
+		setSource file("src/$sourceSetName/java")
```

Comment:
Looking better, can this become a single task with the following as the input?:

```gradle
source sourceSets.main.allSource
source sourceSets.client.allSource
```

Reply:
Wouldn't that include the generated package-info files?

Reply:
Ah yeah, it would do 👍 

I still think you should be able to make this one task though.

Reply:
How about now? :smile:

Reply:
Looks great, thanks 👍